packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -32,13 +35,28 @@ export async function getPageStaticInfo(params: {
   isDev?: boolean
   page?: string
 }): Promise<PageStaticInfo> {
-  const { isDev, pageFilePath, nextConfig } = params
+  const { isDev, pageFilePath, nextConfig, page } = params
 
   const fileContent = (await tryToReadFile(pageFilePath, !isDev)) || ''
   if (/runtime|getStaticProps|getServerSideProps|matcher/.test(fileContent)) {
     const swcAST = await parseModule(pageFilePath, fileContent)
     const { ssg, ssr } = checkExports(swcAST)
-    const config = tryToExtractExportedConstValue(swcAST, 'config') || {}
+
+    // default / failsafe value for config
+    let config: any = {}
+    try {
+      config = extractExportedConstValue(swcAST, 'config')
+    } catch (e) {
+      if (e instanceof UnsupportedValueError) {
+        // `export config` is found, but can't extract its value
+        Log.warn(
+          `You have exported a \`config\` field in "${
+            page || pageFilePath
+          }" that Next.js can't recognize, so it will be ignored`
+        )
+      }
+      // `export config` doesn't exist, or other unknown error throw by swc, silence them
+    }
```

Comment:
Here is what I thought: `extractExportedConstValue` can throw three type of errors:

- `UnsupportedValueError`: Next.js can't extract the value from `config` (E.g., `` { runtime: `experimental-${`edge`}` }``). And I have added a corresponding warning message.
- `NoSuchDeclarationError`: `export config` doesn't exist. In this case, Next.js shouldn't log anything as it is the current and default behavior.
- Other errors: Most likely thrown by swc, may be caused by a `SyntaxError`. Next.js is currently silent (ignore) the warning since we don't know anything about the error. If it is a swc loading error or there is a `SyntaxError` in the file, it will be thrown again during compilation, with more details. So IMHO, we don't need to log any warning in this case.

Reply:
I have thought about adding the logging in `tryToExtractExportedConstValue`. But I'd like to include `pageFilePath` or `route path` in the warning message (the developer can then locate and identify the problematic file).

So I just use `extractExportedConstValue` instead, so that I don't need to pass the context (route path, page file path, etc.) down to `tryToExtractExportedConstValue`.

Reply:
Fair enough. Let's get rid of `tryToExtractExportedConstValue` since it's not used anywhere else.

Reply:
Sure! I have removed it in [`e34ba69` (#38750)](https://github.com/vercel/next.js/pull/38750/commits/e34ba6953f95bac6590436a1d739e6d32e6b4197).